Modules/HMPID/src/HmpidTaskDigits.cxx

```diff
@@ -0,0 +1,181 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file       HmpidTaskDigits.cxx
+/// \author     Antonio Paz, Giacomo Volpe
+/// \brief      Class to map data from HMPID detectors
+/// \version    0.3.1
+/// \date       26/10/2021
+///
+
+//! ap  Changes:
+//     -  Histograms    hHMPIDdigitmapAvg[numCham]   back to   TH2F
+//     -  Histograms    hHMPIDchargeDist[numCham]  back to  TH1F
+
+#include <TCanvas.h>
+#include <TH1F.h>
+#include <TH2F.h>
+#include <TProfile.h>
+
+#include <TMath.h>
+#include <Framework/InputRecord.h>
+
+#include "QualityControl/QcInfoLogger.h"
+#include "HMPID/HmpidTaskDigits.h"
+#include "DataFormatsHMP/Digit.h"
+#include "DataFormatsHMP/Trigger.h"
+
+namespace o2::quality_control_modules::hmpid
+{
+
+HmpidTaskDigits::~HmpidTaskDigits()
+{
+  if (hOccupancyAvg) {

```

Comment:
There is no need anymore to check if it is not a nullptr. `delete nullptr` is valid and is a noop.